modules/ROOT/pages/Development/BeginnersGuide/SimpleMod/machines/SolarPanel.adoc

```diff
@@ -0,0 +1,121 @@
+
+We don't have a building mesh for you to use for the solar panel, but please let us know if you can contribute one!
+For now, just use the Plantpot of Knowledge mesh as a placeholder.
+
+At this point you should be able to construct your buildable in game the same as the Plantpot of Knowledge,
+and just like the plantpot, it doesn't do anything ... yet!
+
+[TIP]
+=====
+There are also subclasses of `FGBuildableGenerator` such as
+`FGBuildableGeneratorFuel`, `FGBuildableGeneratorGeoThermal`, and `FGBuildableGeneratorNuclear`
+which, as you can judge from their names, contain some specific logic for different types of power generators.
+Since our Solar Panel does not require the custom behaviors of any of these, we will not use them.
+=====
+
+== Adding Power Generation Logic
+
+The first thing we want to add to our `Build_DocSolarPanel` is an `FGPowerConnection` component,
+as we need to connect our Solar Panel to power lines to use its generated power.
+The process for adding a component is the same as in the
+xref:Development/BeginnersGuide/SimpleMod/machines/SimpleMachine.adoc[Item Counter Machine] chapter.
+However, unlike the counter, we don't need to add an `FGPowerInfo` this time
+because the `FGBuildableGenerator` class already handles that for us.
+The parent class also already has the code to associate our Power Connection with the Power Info,
+so we don't need to do that ourselves here either.
+
+In order for the Build Gun to properly display the power output of our generator,
+and to define how much power it will make,
+set the "M Power Production" property of the buildable in the Class Defaults.
+Let's go with `20.0`, but you can set it to whatever you'd like.
+
+With the setup so far, we now have a generator that will dynamically produce up to our specified Power Production,
+except it will not charge batteries, and it will still produce power at night.
+That might be okay if we were making an https://en.wikipedia.org/wiki/Radioisotope_thermoelectric_generator[RTG],
+but we're making a solar panel, so we want it to only work during the day.
+
+The first step to fixing this is to disable the generator's dynamic power production ability.
+Use a `Get Power Info` node and call `Set Dynamic Production Capacity` to `0.0`.
+This value defaults to the value we set in our building,
+but setting it to zero here will prevent the undesired functionality
+while still correctly reporting the power production in the build gun.
+Hook up the execution pin to Begin Play, since we only need this to happen once - on actor spawn.
+
+image:BeginnersGuide/simpleMod/machines/SolarPanel/DisableDynamicProduction.png[Setting the Dynamic Production]
+
+Next, we need to figure out whether it's day or night and adjust power production accordingly.
+Thankfully, the `FGTimeOfDaySubsystem` offers some helpful methods for that.
+Get an instance of the `FGTimeOfDaySubsystem` with the `GetTimeSubsystem` node to make use of its `Is Day` method.
```

Comment:
`Is Day` could be mentioned later, it confused me to not see it in the subsequent screenshots